### PR TITLE
Charts loading state

### LINKED ISSUE #66: Show loading state while chart data is being processed for displaying

### LINKED ISSUE #91: Chart staleness (sometimes) gets triggered when changing the time period

### LINKED ISSUE #92: System metrics (CPU/RAM etc) charts don't set the time period correctly sometimes
This only happens when decreasing the time period shown. Example:
- load nook-watcher with time period 1 day -> charts show correct time period (yesterday -> today)
- change time period to 8 days -> charts show correct time period (8 days ago -> today)
- change time period back to 1 day -> charts remain showing the previous time period (8 days ago -> today)

### PR DESCRIPTION
**Pull request by ktkization**
Closes #66 
Closes #91 
Closes #92

### CODE REVIEW

File: webapp/lib/view.dart [hunk 1]
```diff
@@ -549,6 +549,60 @@ class ContentView {
     driverChartsTabContent.children.clear();
   }
 
+  void toggleChartLoadingState(controller.ChartType chartType, bool state, [bool isSystemEvents = false]) {
+    switch(chartType){
+      case controller.ChartType.conversation:
+        if (state) {
+          needsReplyLatestValue.spinner.classes.remove('hidden');
```
Comment (marianamarasoiu) [inline webapp/lib/view.dart]: I would suggest renaming `state` to `show` as it makes it a bit clearer what it means, and then you can use the `toggle()` method to show/hide the `hidden` class based on it, which means that you won't need the if/else statement:
```suggestion
          needsReplyLatestValue.spinner.classes.toggle('hidden', show);
```

File: webapp/lib/view.dart [hunk 2]
```diff
@@ -549,6 +549,60 @@ class ContentView {
     driverChartsTabContent.children.clear();
   }
 
+  void toggleChartLoadingState(controller.ChartType chartType, bool state, [bool isSystemEvents = false]) {
+    switch(chartType){
```
Comment (marianamarasoiu) [inline webapp/lib/view.dart]: nit:
```suggestion
    switch (chartType){
```

**Review by marianamarasoiu — COMMENTED** [review]
LGTM, just a suggestion to replace add/remove with toggle for handling css classes. thanks!

File: webapp/lib/view.dart (hunk 1 above)
Comment (ktkization) [reply webapp/lib/view.dart]: Improved the logic with this

**Review by marianamarasoiu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**